libs/wallet/src/web3-react/connectors/Injected/WidgetEthereumProvider.ts

```diff
@@ -0,0 +1,384 @@
+  targetOrigin?: string
+  // How long to time out waiting for responses. Default 60 seconds.
+  timeoutMilliseconds?: number
+
+  // The event source. By default we use the window. This can be mocked for tests, or it can wrap
+  // a different interface, e.g. workers.
+  eventSource?: Window
+
+  // The event target. By default we use the window parent. This can be mocked for tests, or it can wrap
+  // a different interface, e.g. workers.
+  eventTarget?: Window
+}
+
+/**
+ * This is what we store in the state to keep track of pending promises.
+ */
+interface PromiseCompleter<TResult, TErrorData> {
+  // A response was received (either error or result response).
+  resolve(result: JsonRpcSucessfulResponseMessage<TResult> | JsonRpcErrorResponseMessage<TErrorData>): void
+
+  // An error with executing the request was encountered.
+  reject(error: Error): void
+}
+
+/**
+ * We return a random number between the 0 and the maximum safe integer so that we always generate a unique identifier,
+ * across all communication channels.
+ */
+function getUniqueId(): number {
+  return Math.floor(Math.random() * Number.MAX_SAFE_INTEGER)
+}
+
+export type IFrameEthereumProviderEventTypes =
+  | 'message'
+  | 'connect'
+  | 'disconnect'
+  | 'chainChanged'
+  | 'close' // Deprecated, use 'disconnect' instead
+  | 'notification'
+  | 'networkChanged' // Deprecated, use 'chainChanged' instead
+  | 'accountsChanged'
+
+/**
+ * Export the type information about the different events that are emitted.
+ */
+export interface IFrameEthereumProviderEvents {
+  on(event: 'message', handler: (message: ProviderMessage) => void): this
+
+  on(event: 'connect', handler: (connectInfo: ProviderConnectInfo) => void): this
+
+  on(event: 'disconnect', handler: (error: ProviderRpcError) => void): this
+
+  on(event: 'close', handler: (code: number, reason: string) => void): this
+
+  on(event: 'notification', handler: (result: any) => void): this
+
+  on(event: 'chainChanged', handler: (chainId: string) => void): this
+
+  on(event: 'networkChanged', handler: (networkId: string) => void): this
+
+  on(event: 'accountsChanged', handler: (accounts: string[]) => void): this
+}
+
+/**
+ * Represents an error in an RPC returned from the event source. Always contains a code and a reason. The message
+ * is constructed from both.
+ */
+export class RpcError extends Error {
+  public readonly isRpcError = true
+
+  public readonly code: number
+  public readonly reason: string
+
+  constructor(code: number, reason: string) {
+    super(`${code}: ${reason}`)
+
+    this.code = code
+    this.reason = reason
+  }
+}
+
+/**
+ * This is the primary artifact of this library.
+ */
+export class WidgetEthereumProvider extends EventEmitter<IFrameEthereumProviderEventTypes> {
+  request({ method, params }: { method: string; params: unknown[] }) {
+    // console.log('[IFrameEthereumProvider] IFrameEthereumProvider - rpc request', { method, params })
```

Comment:
Keep or remove?

Reply:
Ok, I will remove but it was like: i find it nice for debugging in the future, but I don't necesarilly want to log it unless I need to debug the RPC calls to the provider outside of the widget.

Reply:
Deleted this and all of the other dead codes for console messages (from this PR and others) https://github.com/cowprotocol/cowswap/pull/3911